bindings/localstorage/localstorage.go

```diff
@@ -0,0 +1,240 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package localstorage
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	securejoin "github.com/cyphar/filepath-securejoin"
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/dapr/pkg/logger"
+	"github.com/google/uuid"
+)
+
+const (
+	fileName = "fileName"
+)
+
+// LocalStorage allows saving files to disk
+type LocalStorage struct {
+	metadata *Metadata
+	logger   logger.Logger
+}
+
+// Metadata defines the metadata
+type Metadata struct {
+	RootPath string `json:"rootPath"`
+}
+
+// NewLocalStorage returns a new LocalStorage instance
+func NewLocalStorage(logger logger.Logger) *LocalStorage {
+	return &LocalStorage{logger: logger}
+}
+
+// Init performs metadata parsing
+func (ls *LocalStorage) Init(metadata bindings.Metadata) error {
+	m, err := ls.parseMetadata(metadata)
+	if err != nil {
+		return err
+	}
+	ls.metadata = m
+
+	err = os.MkdirAll(ls.metadata.RootPath, 0666)
+	if err != nil {
+		return fmt.Errorf("unable to create directory specified by 'rootPath': %s", ls.metadata.RootPath)
+	}
+
+	return nil
+}
+
+func (ls *LocalStorage) parseMetadata(metadata bindings.Metadata) (*Metadata, error) {
+	lsInfo := metadata.Properties
+	b, err := json.Marshal(lsInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	var m Metadata
+	err = json.Unmarshal(b, &m)
+	if err != nil {
+		return nil, err
+	}
+
+	return &m, nil
+}
+
+// Operations enumerates supported binding operations
+func (ls *LocalStorage) Operations() []bindings.OperationKind {
+	return []bindings.OperationKind{
+		bindings.CreateOperation,
+		bindings.GetOperation,
+		bindings.ListOperation,
+		bindings.DeleteOperation,
+	}
+}
+
+func (ls *LocalStorage) create(filename string, req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	d, err := strconv.Unquote(string(req.Data))
+	if err == nil {
+		req.Data = []byte(d)
+	}
+
+	decoded, err := base64.StdEncoding.DecodeString(string(req.Data))
+	if err == nil {
+		req.Data = decoded
+	}
+
+	absPath, _, err := getSecureAbsRelPath(ls.metadata.RootPath, filename)
+	if err != nil {
+		return nil, err
+	}
+
+	err = os.MkdirAll(filepath.Dir(absPath), 0666)
```

Comment:
Can filemask come from metadata too with a 0600 as default?

Reply:
After further investigation, I've set the perms to 0777 by default since the umask will be applied.